src/nodes/chunk_dispatch/chunk_dispatch.c

```diff
@@ -290,8 +362,40 @@ chunk_dispatch_exec(CustomScanState *node)
 	/* Switch to the executor's per-tuple memory context */
 	old = MemoryContextSwitchTo(GetPerTupleMemoryContext(estate));
 
+#if PG15_GE
+	Oid partition_col_idx = InvalidOid;
+	TupleTableSlot *newslot = NULL;
+	if (dispatch->dispatch_state->mtstate->operation == CMD_MERGE)
+	{
+		partition_col_idx =
+			get_partition_attrno_from_tuple(ht->space, slot, state->colnames_list, false);
+		/* Row returned by child node does not have paritition col */
+		if (!OidIsValid(partition_col_idx))
+		{
+			List *actionStates =
+				dispatch->dispatch_state->mtstate->resultRelInfo->ri_notMatchedMergeAction;
+			ListCell *l;
+			foreach (l, actionStates)
+			{
+				MergeActionState *action = (MergeActionState *) lfirst(l);
+				CmdType commandType = action->mas_action->commandType;
+				if (commandType == CMD_INSERT)
+				{
+					/* in case received tuple has only junk attributes, then
+					 * fetch the actual values. */
```

Comment:
why would the received tuple only have junk attributes?

Reply:
Junk attributes, means system column attributes.
If you notice in below plan the tuple recieved by ChunkDispatch has only 2 attributes which are  t.tableoid, t.ctid
From this tuple i cannot know the dimension value. I need to call ExecProject() to get full row and the get the dimension value.

```
postgres=# explain (costs off, verbose) MERGE INTO target t
USING source s
ON t.location = 1234
WHEN NOT MATCHED THEN
INSERT VALUES ('2021-11-01 00:00:05'::timestamp with time zone, 5, 210, 'string - INSERTED BY MERGE');
                                          QUERY PLAN
----------------------------------------------------------------------------------------------
 Custom Scan (HypertableModify)
   ->  Merge on public.target t
         Merge on public.target t_1
         Merge on _timescaledb_internal._hyper_38_13_chunk t_2
         Merge on _timescaledb_internal._hyper_38_14_chunk t_3
         ->  Custom Scan (ChunkDispatch)
               ->  Nested Loop Left Join
                     Output: t.tableoid, t.ctid
                     ->  Append
                           ->  Seq Scan on public.source s_1
                           ->  Seq Scan on _timescaledb_internal._hyper_39_15_chunk s_2
                           ->  Seq Scan on _timescaledb_internal._hyper_39_16_chunk s_3
                     ->  Materialize
                           Output: t.tableoid, t.ctid
                           ->  Append
                                 ->  Seq Scan on public.target t_1
                                       Output: t_1.tableoid, t_1.ctid
                                       Filter: (t_1.location = 1234)
                                 ->  Seq Scan on _timescaledb_internal._hyper_38_13_chunk t_2
                                       Output: t_2.tableoid, t_2.ctid
                                       Filter: (t_2.location = 1234)
                                 ->  Seq Scan on _timescaledb_internal._hyper_38_14_chunk t_3
                                       Output: t_3.tableoid, t_3.ctid
                                       Filter: (t_3.location = 1234)
(24 rows)
```

Reply:
I think iam already doing it.
If you check in chunk_dispatch_plan_create() i have below code.

```
#if PG15_GE
	if (root->parse->mergeUseOuterJoin)
	{
		/* replace expressions of ROWID_VAR */
		tlist = ts_replace_rowid_vars(root, tlist, relopt->relid);
		cscan->scan.plan.targetlist = tlist;
		cscan->custom_scan_tlist = tlist;
	}
#endif
```